matchmaking.py

```diff
@@ -0,0 +1,74 @@
+import random
+import time
+import logging
+
+logger = logging.getLogger(__name__)
+
+
+class Matchmaking:
+    def __init__(self, li, config, username):
+        self.li = li
+        self.variants = config["challenge"]["variants"].copy()

```

Comment:
The next three lines can be written as

    self.variants = list(filter(lambda var: var != "fromPosition", config["challenge"]["variants"]))